test/end2end_test.go

```diff
@@ -4238,6 +4243,73 @@ func testClientRequestBodyErrorCancelStreamingInput(t *testing.T, e env) {
 	})
 }
 
+func TestClientResourceExhaustedCancelFullDuplex(t *testing.T) {
+	defer leakcheck.Check(t)
+	for _, e := range listTestEnv() {
+		if e.httpHandler {
+			// httpHanlder write won't be blocked on flow control window.
+			continue
+		}
+		testClientResourceExhaustedCancelFullDuplex(t, e)
+	}
+}
+
+func testClientResourceExhaustedCancelFullDuplex(t *testing.T, e env) {
+	te := newTest(t, e)
+	recvErr := make(chan error, 1)
+	timeoutErr := errors.New("1s timeout reached")
+	ts := &funcServer{fullDuplexCall: func(stream testpb.TestService_FullDuplexCallServer) error {
+		_, err := stream.Recv()
+		if err != nil {
+			t.Fatalf("expecting <nil> error, got %v", err)
+		}
+		// create a payload that's larger than the default flow control window.
+		payload, err := newPayload(testpb.PayloadType_COMPRESSABLE, 1048576)
+		if err != nil {
+			return err
+		}
+
+		ce := make(chan error)
+		go func() {
+			err := stream.Send(&testpb.StreamingOutputCallResponse{
+				Payload: payload,
+			})
+			ce <- err
+		}()
+		select {
+		case err = <-ce:
+		case <-time.After(time.Second):
```

Comment:
5s or 10s would be better here to help avoid the possibility of a flake caused by a hiccup in travis.  It should only ever be reached in an error case, so we can be pretty generous.